frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/services/JwtServiceTest.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.di.authentication.frontendapi.services;
+
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWEHeader;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.RSADecrypter;
+import com.nimbusds.jose.crypto.impl.ECDSA;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.gen.RSAKeyGenerator;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jwt.EncryptedJWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.kms.model.SignRequest;
+import software.amazon.awssdk.services.kms.model.SignResponse;
+import software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec;
+import uk.gov.di.authentication.shared.services.KmsConnectionService;
+
+import java.nio.charset.StandardCharsets;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class JwtServiceTest {
+    private static final Base64URL TEST_SIGNATURE =
+            new Base64URL(
+                    "DtEhU3ljbEg8L38VWAfUAqOyKAM6-Xx-F4GawxaepmXFCgfTjDxw5djxLa8ISlSApmWQxfKTUJqPP3-Kg6NU1Q");
+    private static final String TEST_CLAIM_NAME = "testClaim";
+    private static final String TEST_CLAIM_VALUE = "helloThere";
+    private static final JWTClaimsSet TEST_CLAIMS =
+            new JWTClaimsSet.Builder().claim(TEST_CLAIM_NAME, TEST_CLAIM_VALUE).build();
+    private static final JWSAlgorithm TEST_ALGORITHM = JWSAlgorithm.ES256;
+    private static final String TEST_KEY_ALIAS = "someSigningKey";
+    private static final Base64URL TEST_EXPECTED_HEADER =
+            new JWSHeader(TEST_ALGORITHM).toBase64URL();
+    private static final RSAKey TEST_KEY_PAIR_GENERATOR = generateTestKeyPair();
```

Comment:
Can we use KeyPairHelper in SharedTest for this?

Reply:
Ah didn't know this existed, have implemented